docs/networking.md

```diff
@@ -0,0 +1,113 @@
+# Networking
+
+Coder's network topology has three types of nodes:
+workspaces, coder servers, and users.
+
+The coder server must have an inbound address reachable by users and workspaces,
+but otherwise, all topologies _just work_ with Coder.
+
+When possible, we establish direct connections between users and workspaces.
+Direct connections are as fast as connecting to the workspace outside of Coder.
+When NAT traversal fails, connections are relayed through the coder server.
+All user <-> workspace connections are end-to-end encrypted.
+
+[Tailscale](https://tailscale.com) backs our networking logic.
```

Comment:
We should be explicit that this isn't Tailscale's SaaS. I think saying "Tailscale's open-source" would resolve it.